tokio/src/runtime/metrics/batch.rs

```diff
@@ -81,8 +113,23 @@ impl MetricsBatch {
         }
 
         let busy_duration = self.last_resume_time.elapsed();
-        let busy_duration = u64::try_from(busy_duration.as_nanos()).unwrap_or(u64::MAX);
-        self.busy_duration_total += busy_duration;
+        self.busy_duration_total += duration_as_u64(busy_duration);
+    }
+
+    pub(crate) fn start_poll(&mut self) {
+        self.poll_count += 1;
+
+        if let Some(poll_timer) = &mut self.poll_timer {
+            poll_timer.poll_started_at = Instant::now();
+        }
+    }
+
+    pub(crate) fn end_poll(&mut self) {
+        if let Some(poll_timer) = &mut self.poll_timer {
+            let elapsed = duration_as_u64(poll_timer.poll_started_at.elapsed());
+            poll_timer.poll_times.measure(elapsed, elapsed);
```

Comment:
This doesn't make sense.

You're adding `elapsed` samples to the bucket that `elapsed` falls within, when really you should just be tracking a single sample.

Continuing from that premise, there's seemingly also no reason to even have a `count` parameter on `measure` given that you're only ever tracking single occurrences.